tensorflow/c/experimental/stream_executor/stream_executor_internal.h

```diff
@@ -83,5 +91,111 @@ class CPlatform : public Platform {
   stream_executor::ExecutorCache executor_cache_;
 };
 
+class CStream : public internal::StreamInterface {
+ public:
+  CStream(SP_Device* device, SP_StreamExecutor* stream_executor)
+      : device_(device),
+        stream_executor_(stream_executor),
+        stream_handle_(nullptr) {}
+  ~CStream() override { Destroy(); }
+
+  port::Status Create() {
+    OwnedTFStatus c_status(TF_NewStatus());
+    stream_executor_->create_stream(device_, &stream_handle_, c_status.get());
+    port::Status s = tensorflow::StatusFromTF_Status(c_status.get());
+    return s;
+  }
+
+  void Destroy() {
+    if (stream_handle_ != nullptr) {
+      stream_executor_->destroy_stream(device_, stream_handle_);
+      stream_handle_ = nullptr;
+    }
+  }
+
+  SP_Stream Handle() { return stream_handle_; }
+
+ private:
+  SP_Device* device_;
+  SP_StreamExecutor* stream_executor_;
+  SP_Stream stream_handle_;
+};
+
+class CEvent : public internal::EventInterface {
```

Comment:
Why did you have to move this class?

Reply:
originally it was in stream_executor.cc, however, we need to access its member function in other file(kernel.cc::TF_GetStream), so I moved it to the header, or other module cannot access its member.
```
SP_Stream TF_GetStream(TF_OpKernelContext* ctx) {
  auto* cc_ctx = reinterpret_cast<::tensorflow::OpKernelContext*>(ctx);
  if (cc_ctx->op_device_context() == nullptr) {  // CPU Device
    return nullptr;
  } else if (!cc_ctx->op_device_context()->IsPluggableDevice()) {
    return nullptr;
  } else {  // Is a PluggableDevice
    auto c_stream = static_cast<stream_executor::CStream*>(
        cc_ctx->op_device_context()->stream()->implementation());
    return c_stream->Handle();
  }
}

```

Reply:
I'd suggest splitting out NFC code movement changes like these into a separate PR.  The NFC changes will land quickly and these non-NFC changes will also be simpler to review.

Reply:
@annarev @sanjoy @penpornk I have prepared a new PR[#44747](https://github.com/tensorflow/tensorflow/pull/44747])which moves those kernel/stream executor C API changes(kernels.cc, kernels.h, kernels_test, stream_executor.cc, stream_executor) from this PR, what's your preference, continue in this PR or a separate PR? If a separate PR is preferred, I will remove those changes in this PR. Thanks.

Reply:
@annarev @sanjoy @penpornk I have moved these kernels/stream executor C API changes to PR[#44747](https://github.com/tensorflow/tensorflow/pull/44747) and removed those changes from this PR, please have a review. Thanks